### PR TITLE
authCV initialized to null, getter used in checkAuth

### PR DESCRIPTION
**Pull request by NunoDasNeves**
idk how to enable prerender on staging - it seems like it's enabled everywhere according to `src/backend/index.js` ??

### CODE REVIEW

File: frontend/src/store/auth/index.js [hunk 1]
```diff
@@ -27,7 +27,10 @@ const getters = {
   userAuthObject: ({ userAuthObject }) => userAuthObject,
   loading: ({ loading }) => loading,
   error: ({ error }) => error,
-  authCV: ({ authCV }) => authCV
+  authCV: ({ authCV }) => {
```
Comment (DarkPurple141) [inline frontend/src/store/auth/index.js]: This doesn't work. It'll just alter the param, not the underlying CV.

I suggest this:

```js
const _authCV = new CV()

const state = {...}

const getters = { authCV: () => _authCV }
const mutations = { ...SIGNAL_AUTH_CV() { _authCV.signal() }}
```

File: frontend/src/store/auth/index.js [hunk 2]
```diff
@@ -12,7 +12,7 @@ const state = {
   // our own user data
   profile: null,
   // condition variable for app to wait on while waiting for auth to resolve on boot
-  authCV: new CV()
+  authCV: null
```
Comment (DarkPurple141) [inline frontend/src/store/auth/index.js]: and then just remove this.

File: frontend/src/store/auth/index.js [hunk 3]
```diff
@@ -193,7 +193,7 @@ const actions = {
 
     // signal the CV so the app can continue loading and use the JWT token in its requests
     // Note we _need_ to do this before returning!
-    commit('SIGNAL_AUTH_CV')
```
Comment (DarkPurple141) [inline frontend/src/store/auth/index.js]: And then retain this as a commit

File: frontend/src/store/auth/index.js (hunk 1 above)
Comment (NunoDasNeves) [reply frontend/src/store/auth/index.js]: can't i just not destructure state? that seems to work
```
  authCV: (state) => {
    if (state.authCV === null) state.authCV = new CV()
    return state.authCV
  }
```

Comment (NunoDasNeves) [reply frontend/src/store/auth/index.js]: Also this is why we should be using Rust for our frontend ^_^

**Comment by NunoDasNeves** [comment]
> I mean I guess. Seems like violation of getters being non mutating. Also not sure if authCV is really application state.

hok then

**Comment by DarkPurple141** [comment]
@NunoDasNeves also the previous deploy was broken so 👍

File: frontend/vue.config.js [hunk 4]
```diff
@@ -15,10 +17,26 @@ const productionPlugins = [
       // so we create a snapshot of the page after the
       // `data-view` attribute exists in the DOM.
       renderAfterElementExists: '[data-view]'
-    })
+    }),
+    postProcess (renderedRoute) {
+      // Ignore any redirects.
+      renderedRoute.route = renderedRoute.originalRoute
+      // Basic whitespace removal. (Don't use this in production.)
```
Comment (NunoDasNeves) [inline frontend/vue.config.js]: don't use in production? o.O

File: frontend/vue.config.js [hunk 5]
```diff
@@ -15,10 +17,26 @@ const productionPlugins = [
...
+      renderedRoute.html = renderedRoute.html.split(/<div id="app"/)
```
Comment (NunoDasNeves) [inline frontend/vue.config.js]: this has been added to index.html already

Comment (DarkPurple141) [reply frontend/vue.config.js]: Oh this is a quote I copy-pasta'd. Just wanted to see if this works to stop re-render.

File: frontend/.env.production [hunk 6]
```diff
@@ -1,3 +1,4 @@
 NODE_ENV=production
 VUE_APP_TITLE=SmartCourse
-VUE_APP_API_URL=https://smartcourse.me/api
\ No newline at end of file
+VUE_APP_API_URL=https://smartcourse.me/api
+VUE_APP_URL=https://smartcourse-staging.azurewebsites.net
```
Comment (NunoDasNeves) [inline frontend/.env.production]: uhh is this right?

File: frontend/.env.staging [hunk 7]
```diff
@@ -1,3 +1,4 @@
 NODE_ENV=staging
 VUE_APP_TITLE=SmartCourse (staging)
 VUE_APP_API_URL=https://smartcourse-staging.azurewebsites.net/api
+VUE_APP_URL=https://smartcourse.me
```
Comment (NunoDasNeves) [inline frontend/.env.staging]: i think you got these switcharoo'd

**Pull request merged**